adafruit_minimqtt/adafruit_minimqtt.py

```diff
@@ -194,11 +194,12 @@ def __init__(
 
         self.port = MQTT_TCP_PORT
         if is_ssl:
+            self._is_ssl = True
```

Comment:
As written this argument doesn't actually work if user attempts to set it to `False` In that case ` if is_ssl:` evaluates to False and therefore `self._is_ssl` never gets created and there is a exception raised when the code first tries to access it. 

I tested one possible fix by changing this line and moving it to above the if statement like this:
```
        self._is_ssl = is_ssl
        self.port = MQTT_TCP_PORT
        if is_ssl:
            self.port = MQTT_TLS_PORT
        if port:
            self.port = port
```
This way `self._is_ssl` will always get created and set to the value that the user passed in.

Reply:
Thanks ! I used that as a basis, hopefully it looks better now.

Reply:
Also, I added buch of tests that will hopefully make it more clear.

Reply:
With the changed code, there is a change of behavior - the `is_ssl` has to be specified explicitly even for the default TLS port. Should I make port 8883 special as well ?